### ISSUE TITLE
:sparkles: Support Request Dependencies

### ISSUE THREAD

**Issue by tombulled** [neoclient/neoclient#31]
For example, to greet a random name:
```python
def random_name() -> str:
    return random.choice(["sam", "bob", "sally"])

@get("/greet")
def greet(name: str = Depends(random_name)):
    ...
```

Or:
```python
def random_name() -> str:
    return random.choice(["sam", "bob", "sally"])

def my_dependency(name: str = Depends(random_name)):
    ...

@get("/greet", dependencies=[my_dependency])
def greet():
    ...
```

Or:
```python
def random_name() -> str:
    return random.choice(["sam", "bob", "sally"])

@query("name", factory=random_name)
@get("/greet", dependencies=[my_dependency])
def greet():
    ...
```

**Comment by tombulled** [comment]
The general use case here is a reusable set of query params/headers etc. that multiple requests all need.
For example:
```python
@header("x-client", "PCLI")
@query("format", "json")
def foo():
    ...

@header("x-client", "PCLI")
@query("format", "json")
def bar():
    ...
```
Middleware would be an obvious solution for this

**Comment by tombulled** [comment]
TODO: Merge this issue with #73

**Comment by tombulled** [comment]
Closed as duplicate of #73
